### ISSUE TITLE
Canon 80D not capturing

### ISSUE THREAD

**Issue by uzelessknowledge** [gphoto/gphoto2#68]
When I do a --auto-detect I get back USB PTP Class Camera for my 80D. I can't do a capture with it. 
New to this so not sure what is needed to assist.
Thanks

**Comment by msmeissn** [comment]
what does:
gphoto2 --version
report?

Likely libgphoto2 is too old.

**Comment by uzelessknowledge** [comment]
gphoto2 2.5.4
Libgphoto2 2.5.4
Libgphoto2_port 0.10.0

I ran sudo apt-get install gphoto2. I see in the release notes I need version 13. How do I update to another version?

**Comment by msmeissn** [comment]
This is too old sadly.  2.5.13 is probably best used.

Check out https://github.com/gonzalo/gphoto2-updater   for an easy gphoto2/libgphoto2 update script...

**Comment by uzelessknowledge** [comment]
That script ran, but after checking the version it still showed 2.5.4. Not sure why it showed it installed 2.5.11, but didn't.
I'm running Pixel on my Mac to figure this out so I can try it on the Pi. I had to manually install libgphoto2 and gphoto2. 
Installed libgphoto2 2.5.13 and gphoto 2.5.11. If I did it right. When I check the version I see.

_This version of gphoto2 is using the following software versions and options:
gphoto2         2.5.11         gcc, popt(m), exif, no cdk, no aa, no jpeg, no readline
libgphoto2      2.5.4          all camlibs, gcc, ltdl, EXIF
libgphoto2_port 0.12.0         gcc, ltdl, USB, serial without locking_

Why does libphoto2 still show 2.5.4?
[output.txt](https://github.com/gphoto/gphoto2/files/959671/output.txt)

**Comment by uzelessknowledge** [comment]
Ran the script on the pi tonight and have the same result.
I still can't get libgphoto2 to update. 
Even tried starting with a fresh Pixel install on the SD.

**Comment by uzelessknowledge** [comment]
Ok this is strange, but worked. I thought about trying to remove libgphoto2 and suddenly gphoto2 --version shows libgphoto2 as 2.5.12. 
Gphoto2 --auto-detect shows the camera. 

I ran sudo apt-get remove libgphoto2*. 
Maybe something needs to be updated in the install process?

**Comment by msmeissn** [comment]
ok, as it now shows 2.5.12 .... does the 80D work?

**Comment by uzelessknowledge** [comment]
Yes. Seemed to with a quick test and then I went to bed. This is on a test pi and not the one I'll be using so trying to figure out the kinks. 
I'm working to do a fresh install on Jessie and start over. 
Thank you for the help. I'll report back.

**Comment by uzelessknowledge** [comment]
OK. So I tried a fresh image and install.  Needed some dependencies. Took a bit to figure those out. 
1. Install libltdl-dev libusb-dev libusb-1.0 libexif-dev libpopt-dev
2. Install lbgphoto2 from the source. v.2.5.13
3. Install Gphoto2 from source v2.5.11
4. Check Gphoto version - version is 2. 5.11, but libgphoto is 2.5.4
5. sudo apt-get remove libgphoto2*
6. Check version again and this time Gphoto is 2.5.11 and libgphoto is 2.5.13

I won't be able to test with the camera until later.
Am I supposed to have to remove libphoto2 or am I compiling something wrong?

**Comment by msmeissn** [comment]
The gphoto2-updater.sh script usually installs into /usr/local to avoid overwriting the system installation.

so to activate it the system libgphoto2 would need to be removed, or /usr/local/bin/gphoto2 to be called.

if you build yourself, pass --prefix=/usr  to configure (and potentially --libdir=/usr/..... whereever the libraries live)   to then later overwrite the system libraries. 

2.5.12 should have been OK already. so you should be good ;)

**Comment by uzelessknowledge** [comment]
Hmm. That just surpassed what I know. Seems I'm good this way. We'll see. Really appreciate the help.

**Comment by uzelessknowledge** [comment]
OK. started with a fresh install of Jessie. I installed gphoto2 with the script (which has been updated in git) you provided above. Also installed INDI server. 
--auto-detect does see the camera, but when I try to run --trigger-capture i get an error.

pi@raspberrypi:~ $ gphoto2 --auto-detect
Model                          Port                                            
----------------------------------------------------------
Canon EOS 80D                  usb:001,006     
pi@raspberrypi:~ $ gphoto2 --trigger-capture
ERROR: Could not trigger capture.                                              
*** Error (-7: 'I/O problem') ***    


pi@raspberrypi:~ $ gphoto2 --version
gphoto2 2.5.11

Copyright (c) 2000-2016 Lutz Mueller and others

gphoto2 comes with NO WARRANTY, to the extent permitted by law. You may
redistribute copies of gphoto2 under the terms of the GNU General Public
License. For more information about these matters, see the files named COPYING.

This version of gphoto2 is using the following software versions and options:
gphoto2         2.5.11         gcc, popt(m), exif, no cdk, no aa, jpeg, no readline
libgphoto2      2.5.13         all camlibs, gcc, ltdl, EXIF
libgphoto2_port 0.12.0         gcc, ltdl, USB, serial without locking
pi@raspberrypi:~ $

**Comment by uzelessknowledge** [comment]
I loaded libgphoto2 2.5.12 and I was able to capture. So it would seem .13 doesn't like the 80D.
To confirm .13 was bad I tried on 2 different SD cards and also in my VM. I reverted 1 SD and my VM and both work fine on .12. 
Now to get INDI working...tomorrow. Good night.

**Comment by msmeissn** [comment]
I was hoping not to have regressed it in 2.5.13 . and i see no indications that I did. if possible, a debug logfile with 2.5.13 would be good

**Comment by uzelessknowledge** [comment]
Hey. I can grab some logs, although hate to undo what I have working. Can you point me in the direction of them? 
I've now got gphoto2 working and Indi Server working thanks to scribblemaniac.

**Comment by uzelessknowledge** [comment]
Logs and screen shots are attached. I ran this in both 2.5.12 and 2.5.13 on the same VM with no reboot or change other than installing 2.5.13 and then 2.5.12. Let me know what else I can provide.
[Gphoto Issue 68.zip](https://github.com/gphoto/gphoto2/files/969217/Gphoto.Issue.68.zip)

**Comment by msmeissn** [comment]
The 2.5.13 bug in your logfile is a shuttercount query which resuilts in error, I fixed this in current GIT some days ago.

the 2.5.12 log also has an issue :/

**Comment by msmeissn** [comment]
commit 6779033c90abcac142bd2e747e35daa065ee04ba
